libpod/options.go

```diff
@@ -2203,3 +2203,21 @@ func WithPodInfraExitCommand(exitCmd []string) PodCreateOption {
 		return nil
 	}
 }
+
+// WithPodSlirp4netns tells the pod to use slirp4netns.
+func WithPodSlirp4netns(networkOptions map[string][]string) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+
+		if !pod.config.InfraContainer.HasInfraContainer {
+			return errors.Wrapf(define.ErrInvalidArg, "cannot configure pod networking as no infra container is being created")
+		}
+
+		pod.config.InfraContainer.Slirp4netns = true
```

Comment:
Can you add a check to see if `pod.config.InfraContainer.HostNetwork`  is set already? We should only allow one of `Slirp4netns` and `HostNetwork` to be set.